Add markdownlint CI

pbricout opened the pull request:
Add markdown linter as part of the CI.
See:
https://github.com/DavidAnson/markdownlint

Zazama commented:
Looks good to me, I think I'm just gonna move all tests to GitHub CI for simplicity.

One thing: When I try to run the command in the package.json on Windows, it won't work because of the `'**/*.md'` escape. When I change it to just `**/*.md` it works fine. Are the `''` needed for you?

pbricout commented:
> Looks good to me, I think I'm just gonna move all tests to GitHub CI for simplicity.

Could you keep the tests in Scrutinizer for now, I would like to use Scrutinizer to improve the code quality based on these reports:
https://scrutinizer-ci.com/g/Zazama/node-id3/code-structure/master/hot-spots

> One thing: When I try to run the command in the package.json on Windows, it won't work because of the '**/*.md' escape. When I change it to just **/*.md it works fine. Are the '' needed for you?

Unfortunately, the command needs to be different for Windows and Unix:
https://github.com/igorshubovych/markdownlint-cli#globbing-examples

I am not sure how to best deal with it ?

We could have two scripts `markdownlint-windows` and `markdownlint-unix` thoughts?

pbricout commented on .circleci/config.yml:
```diff
@@ -9,22 +9,34 @@ commands:
       - image: node:10
     steps:
       - test-node-id3
+      - markdownlint
```
This does not seem to be taken into account (yet)? Do you have any idea why?

pbricout replied:
Ah I can see an error in Circle CI now. `scrutinizer` seems to be a different tool.

pbricout replied:
```text
ERROR IN CONFIG FILE:
[#/commands/markdownlint] 0 subschemas matched instead of one
1. [#/commands/markdownlint] expected type: String, found: Mapping
|   Command may be a string reference to another command
2. [#/commands/markdownlint/steps] 10 schema violations found
```

I might seem simply remove it, thoughts?

pbricout commented:
The gihub CI workflow does not seem to work, you might need to enable for this repo in the `Actions` tab, see:
https://docs.github.com/en/actions/using-workflows/using-starter-workflows

pbricout commented:
> Looks good to go for me, or is something missing?

Thanks for migrating to Github CI, I rebased and cleaned the commits, looks good to me now, thanks.

Zazama commented:
Great, thanks 🙂
